pysteps/nowcasts/lagrangian_probability.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+"""
+pysteps.nowcasts.lagrangian_probability
+======================================
+
+Implementation of Lagrangian probability nowcasting methods.
+
+.. autosummary::
+    :toctree: ../generated/
+
+    forecast
+"""
+
+import numpy as np
+from scipy import signal
+
+from pysteps.nowcasts import extrapolation
+
+
+def forecast(
+    precip,
+    velocity,
+    timesteps,
+    threshold,
+    extrap_method="semilagrangian",
+    extrap_kwargs=None,
+    slope=5,
+):
+    """
+    Generate a probability nowcast by a local lagrangian approach.
+
+    Parameters
+    ----------
+    precip: array-like
+       Two-dimensional array of shape (m,n) containing the input precipitation
+       field.
+    velocity: array-like
+       Array of shape (2,m,n) containing the x- and y-components of the
+       advection field. The velocities are assumed to represent one time step
+       between the inputs.
+    timesteps: int or list of floats
+       Number of time steps to forecast or a list of time steps for which the
+       forecasts are computed (relative to the input time step). The elements of
+       the list are required to be in ascending order.
+    threshold: float
```

Comment:
mmmh convolution would still need to be done independently for each threshold, but you would need to perform only one extrapolation forecast. I can try to add the option!

Reply:
edit: not that convinced that it would bring much benefit, with the risk of cluttering the code... I'd prefer leaving this feature out for the time being.